great_expectations/rule_based_profiler/parameter_builder/multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,131 @@
+    or obtained from the configured BatchRequest parameter).  The reason for this design is that parameter naming is
+    specific to the convention between the ParameterBuilder and ExpectationConfigurationBuilder employed in a specific
+    use-case (i.e., parameter names are created with the ultimate usage by the ExpectationConfigurationBuilder in mind).
+    """
+
+    def __init__(
+        self,
+        parameter_name: str,
+        data_context: Optional[DataContext] = None,
+        batch_request: Optional[Union[dict, str]] = None,
+    ):
+        """
+        Args:
+            parameter_name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        if data_context is None:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"{self.__class__.__name__} requires a data_context, but none was provided."
+            )
+
+        super().__init__(
+            parameter_name=parameter_name,
+            data_context=data_context,
+        )
+
+        self._batch_request = batch_request
+
+    @abstractmethod
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        validator: Validator,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
+        pass
+
+    def get_batch_ids_for_metrics_calculations(
+        self,
+        domain: Optional[Domain] = None,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ) -> Optional[List[str]]:
+        if self._batch_request is None:
+            return None
+
+        # Obtain BatchRequest from rule state (i.e., variables and parameters); from instance variable otherwise.
+        batch_request: Optional[
+            Union[BatchRequest, dict, str]
+        ] = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=self._batch_request,
+            expected_return_type=dict,
+            variables=variables,
+            parameters=parameters,
+        )
+        batch_request = BatchRequest(**batch_request)
+        batch_list: List[Batch] = self.data_context.get_batch_list(
+            batch_request=batch_request
+        )
+
+        batch: Batch
+        batch_ids: List[str] = [batch.id for batch in batch_list]
+
+        return batch_ids
+
+    def get_validator_for_metrics_calculations(
```

Comment:
Why do we have this method for this parameter_builder, but not for others?

Reply:
@talagluck This is a great question.  For this PR, it is only needed for the multi-batch case.  For the next PR to be associated with `https://superconductive.atlassian.net/browse/GE-297`, we will probably have it widely available.

Reply:
To confirm: before we didn't need an additional Validator because we were only working with the Validator that was passed in?

Reply:
Yes, and in the future, we will have the ability to obtain `batch_ids` and/or `Validator` -- independently for `DomainBuilder` and `ParameterBuilder`.  In the case of the `DomainBuilder`, the likely use will be to obtain columns and other metadata for the explicit purposes of building `Domain` objects.  In the case of the `ParameterBuilder`, it will be mostly for extensive metrics computations.  Moreover, analogously to how the functions that obtain `batch_ids` and the `validator` are defined in the higher level `MultiBatchParameterBuilder` for the computational purposes utilized by `NumericMetricRangeMultiBatchParameterBuilder` in the present PR, in the near future, these utilities (i.e., these methods might be migrated to general utility modules or available in the `Profiler` or `Rule` classes themselves, rather than reside in a superclass) will be available for use by both `DomainBuilder` classes and `ParameterBuilder` classes.